clipper_admin/clipper_manager.py

```diff
@@ -628,13 +630,136 @@ def _save_python_function(self, name, predict_function):
+                    ignore_list.append(get_module_name(line))
+
+        if (os.path.isfile(pip_deps_fname)):
+            with open(pip_deps_abs_path, 'r') as f:
+                for line in f:
+                    ignore_list.append(get_module_name(line))
+
+        with open(python_container_conda_deps_fname, 'r') as f:
+            for line in f:
+                ignore_list.append(line.strip())
+
+        return ignore_list
+
+    def _export_local_modules(self, func_modules, serialization_dir,
+                              conda_deps_fname, pip_deps_fname,
+                              local_modules_folder_name):
+        """
+        Supplies modules in `func_modules` not already accounted for by Anaconda or pip to container.
+        Copies those modules to {`serialization_dir`}/{`local_module_deps_fname`}.
+
+        Parameters
+        ----------
+        func_modules : str
+            The modules to consider exporting
+        serialization_dir : str
+            The path to the diretory containing the conda, pip, and output local modules dependency information
+        conda_deps_fname : str
+            The name of the conda dependency file
+        pip_deps_fname : str
+            The name of the pip dependency file
+        local_modules_folder_name : str
+            The name of the folder to copy modules to
+        """
+        ignore_list = self._get_already_exported_modules(
+            serialization_dir, conda_deps_fname, pip_deps_fname)
+        paths_to_copy = set()
+
+        mda = ModuleDependencyAnalyzer()
+        for module_name in ignore_list:
+            mda.ignore(module_name)
+        for module in func_modules:
+            module_name = module.__name__
+            if module_name not in ignore_list and 'clipper_admin' not in module_name:
+                mda.add(module_name)
+
+                if '.' in module_name:
+                    module_name_split = module_name.split()
+                    try:
+                        file_location = module.__file__
+                        package_nested_layer = len(module_name.split('.')) - 1
+                        folder_indices = [
+                            i for i, x in enumerate(file_location) if x == "/"
+                        ]
+                        cutoff_index = folder_indices[-package_nested_layer]
+                        package_path = file_location[:cutoff_index]
+                        paths_to_copy.add(package_path)
+                    except:
+                        print(
+                            "Could not identify the location of the root package {package} for module {module}. Will not supply it to the container.".
+                            format(
+                                package=module_name_split[0],
+                                module=module_name))
+            else:
+                mda.ignore(module_name)
+        module_paths = mda.get_and_clear_paths()
+
+        def path_already_captured(path):
```

Comment:
We discussed this in person, but to re-iterate, let's keep the de-duplication logic (that seems useful).

Let's support the case when users import packages but not try to support when users add the enclosing folder to `sys.path`. That's sort of an anti-pattern in Python (despite the fact we make occasional use of it for good reasons within this repo) and it's okay to not support it.